scss/common/_base.scss

```diff
@@ -418,59 +418,63 @@
         opacity: .3;
     }
 
-    $loading-stroke-width: 2px !default;
-    $loading-size: $icon-size - $loading-stroke-width !default;
-    $loading-speed: 2s !default;
-    $loading-color: $accent;
-    $loading-outer-size: ($loading-size + ($loading-stroke-width * 2));
-    $loading-half-size: ($loading-outer-size / 2) !default;
-
-    .k-icon.k-i-loading {
+    // Loading indicator
+    $loading-size: 1em !default;
+    $loading-base-font-size: 16 !default;
+    $loading-base-font-size-px: 16px !default;
+    $loading-font-size: 16 !default;
+    $loading-font-size-px: 16px !default;
+    $loading-stroke-width: calc( .03 *( #{$loading-font-size} / #{$loading-base-font-size}) * 1px) !default;
+
+    .k-i-loading {
         position: relative;
-        width: $loading-size;
-        height: $loading-size;
-        border: 2px solid $bg-color;
-        box-shadow: inset 0 0 0 $loading-stroke-width $loading-color;
-        border-radius: 50%;
-        box-sizing: content-box;
-
-        &::before,
-        &::after {
-            content: "";
-        }
+        background-color: transparent;
+        box-sizing: border-box;
+        font-size: $loading-font-size-px;
+        color: currentColor;
     }
 
     .k-i-loading::before,
     .k-i-loading::after {
         position: absolute;
-        width:  $loading-half-size;
-        height: $loading-outer-size;
-        background: $bg-color;
+        top: 50%;
+        left: 50%;
+        display: inline-block;
+        content: "";
+        box-sizing: inherit;
+        border-radius: 50%;
+        border-width: $loading-stroke-width;
+        border-style: solid;
+        border-color: currentColor;
+        border-top-color: transparent;
+        border-bottom-color: transparent;
+        background-color: transparent;
     }
+
     .k-i-loading::before {
-        border-radius: $loading-outer-size 0 0 $loading-outer-size;
-        top: -($loading-stroke-width);
-        left: -($loading-stroke-width);
-        transform-origin: $loading-half-size $loading-half-size;
-        animation: k-loading-animation $loading-speed infinite ease 1.5s;
+        margin-top: -($loading-size / 2);
+        margin-left: -($loading-size / 2);
+        width: $loading-size;
+        height: $loading-size;
+        animation: rotate .7s linear infinite;
     }
+
     .k-i-loading::after {
-        border-radius: 0 $loading-outer-size $loading-outer-size 0;
-        top: -($loading-stroke-width);
-        left: ($loading-size / 2);
-        transform-origin: 0 $loading-half-size;
-        animation: k-loading-animation $loading-speed infinite ease;
+        margin-top: -($loading-size / 4);
+        margin-left: -($loading-size / 4);
+        width: $loading-size / 2;
+        height: $loading-size / 2;
+        animation: rotate reverse 1.4s linear infinite;
     }
 
-    @keyframes k-loading-animation {
+    @keyframes rotate  {
```

Comment:
This may clash with animation names defined by the customer. Why not keep it `k-loading-animation`?